provider/awssd/aws_sd.go

```diff
@@ -252,18 +306,273 @@ func (p *AWSSDProvider) updatesToCreates(changes *plan.Changes) (creates []*endp
 	for _, old := range changes.UpdateOld {
 		current := updateNewMap[old.DNSName]
 
+		// always register (or re-register) instance with the current data
+		creates = append(creates, current)
+
 		if !old.Targets.Same(current.Targets) {
 			// when targets differ the old instances need to be de-registered first
 			deletes = append(deletes, old)
+			// remove from both current.Targets and create.Targets the targets that appear in both of them
+			p.DedupDeletesAndCreates(current, old)
```

Comment:
How does this happen? Shouldn't `current` and `old` be always different when it's update?

Reply:
`current.Targets` and `old.Targets` are always different, but they may have common items. For example, when a node is added to the k8s cluster, `old` contains all the existing nodes and `current` contains all the existing nodes plus the new one. This is an issue, because external-dns will try to de-register and then re-register the existing nodes (causing unneeded calls that overlap and result in errors). `DedupDeletesAndCreates` solves the issue by removing all items from `old` and by removing the same items from `current` (`current` then only contains the new node.).